### DISCUSSION TITLE
What means loop will block for I/O?

### DISCUSSION THREAD

**Discussion by bang9** [libuv/libuv#3256] [Q&A]
Hi, I have a question in poll phase
I read about many articles and libuv [designs](http://docs.libuv.org/en/v1.x/design.html) related to event loops.

In poll phase
1. timeout is calculated
2. loop will block I/O by timeout

In step 2, is it I/O blocking operations? if not, hold the loop in the poll phase for waiting I/O events and trigger callback by specified timeout?

Thanks.

**Comment by santigimeno** [comment]
It means that the platform-specific polling method will block waiting for I/O operations a maximum time of the `timeout` calculated in 1. The timeout value can be:
- 0, meaning it'll return immediately whether there a I/O operations or not.
- `> 0`, meaning it'll return either when a I/O operation is ready or the `timeout` is due.
- Infinity, meaning it'll return only when a I/O op is ready.

**Reply by bang9** [reply]
Thank you for your answer!
Is the block means limiting the working time of the poll phase?
